packages/hardhat-core/src/internal/core/providers/construction.ts

```diff
@@ -77,13 +77,14 @@ export function createProvider(
     );
 
     eip1193Provider = new HardhatNetworkProvider(
-      hardhatNetConfig.hardfork!,
+      hardhatNetConfig.hardfork,
       HARDHAT_NETWORK_NAME,
-      hardhatNetConfig.chainId!,
-      hardhatNetConfig.chainId!,
-      hardhatNetConfig.blockGasLimit!,
-      hardhatNetConfig.throwOnTransactionFailures!,
-      hardhatNetConfig.throwOnCallFailures!,
+      hardhatNetConfig.chainId,
+      hardhatNetConfig.chainId,
+      hardhatNetConfig.blockGasLimit,
+      hardhatNetConfig.minGasPrice,
+      hardhatNetConfig.throwOnTransactionFailures,
+      hardhatNetConfig.throwOnCallFailures,
```

Comment:
The only important change here is that `minGasPrice` is a new argument. The other line changes are unrelated to this PR, it's just that all these non-null assertions weren't necessary. They were probably a leftover from the previous config types.

Reply:
Thanks for the clarification. 

I wonder if there's a tslint (eslint?) rule to forbid unnecessary `!`s.